test/net/sourceforge/kolmafia/textui/parsetree/ValueTest.java

```diff
@@ -33,167 +31,143 @@ void compareToNull() {
           hasSign(NEGATIVE));
-      assertThat(
-          "buffers compare to floats by float value",
-          value.compareTo(new Value(-30.4)),
-          hasSign(POSITIVE));
+      assertThat("\"contents\" > -30.4", value.compareTo(new Value(-30.4)), hasSign(POSITIVE));
     }
 
     @Test
     void compareClasses() {
+      // classes compare as ids
       var value = Objects.requireNonNull(DataTypes.makeClassValue(4, true));
       assertThat(
-          "classes compare to classes by int value (different value)",
+          "$class[4] > $class[3]",
           value.compareTo(DataTypes.makeClassValue(3, true)),
           hasSign(POSITIVE));
       assertThat(
-          "classes compare to classes by int value (same value)",
+          "$class[4] == $class[3]",
           value.compareTo(DataTypes.makeClassValue(4, true)),
           hasSign(ZERO));
+      assertThat("$class[4] > 2.4", value.compareTo(new Value(2.4)), hasSign(POSITIVE));
+      assertThat("$class[4] > 2", value.compareTo(new Value(2)), hasSign(POSITIVE));
       assertThat(
-          "classes compare to floats by float value",
-          value.compareTo(new Value(2.4)),
-          hasSign(POSITIVE));
-      assertThat(
-          "classes compare to ints by int value", value.compareTo(new Value(2)), hasSign(POSITIVE));
-      assertThat(
-          "classes compare to monsters by int value",
+          "$class[4] < $monster[99]",
           value.compareTo(DataTypes.makeMonsterValue(99, true)),
           hasSign(NEGATIVE));
     }
 
     @Test
     void compareCoinmasters() {
+      // coinmasters compare as names, or int 0
       var value =
           Objects.requireNonNull(DataTypes.makeCoinmasterValue(DiscoGiftCoRequest.DISCO_GIFTCO));
       assertThat(
-          "coinmasters compare to coinmasters by string value (different value)",
+          "$coinmaster[Disco GiftCo] < $coinmaster[Niña Store]",
           value.compareTo(DataTypes.makeCoinmasterValue(NinjaStoreRequest.NINJA_STORE)),
           hasSign(NEGATIVE));
       assertThat(
-          "coinmasters compare to coinmasters by string value (same value)",
+          "$coinmaster[Disco GiftCo] < $coinmaster[Disco GiftCo]",
           value.compareTo(DataTypes.makeCoinmasterValue(DiscoGiftCoRequest.DISCO_GIFTCO)),
           hasSign(ZERO));
       assertThat(
-          "coinmasters compare to floats by float value",
-          value.compareTo(new Value(2.4)),
-          hasSign(NEGATIVE));
+          "$coinmaster[Disco GiftCo] < 2.4", value.compareTo(new Value(2.4)), hasSign(NEGATIVE));
+      assertThat("$coinmaster[Disco GiftCo] < 2", value.compareTo(new Value(2)), hasSign(NEGATIVE));
       assertThat(
-          "coinmasters compare to ints by int value",
-          value.compareTo(new Value(2)),
-          hasSign(NEGATIVE));
-      assertThat(
-          "coinmasters compare to monsters by int value",
+          "$coinmaster[Disco GiftCo] < $monster[453]",
```

Comment:
Honestly this needs to just be not allowed. It needs to throw an error or something

Reply:
Seeing it in pseudocode makes me realise.

Reply:
In Mafia (ASH), most of these aren't allowed -- you get an error about not being able to compare types.

Reply:
Right ok